### PR TITLE
Plugin example

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
**Ready to merge**
Create an example of plugin for early plugin developers
- [X] Listen on compiler
- [X] Call the FileSystem
- [x] Use Bootstrap style

### CODE REVIEW

File: examples/plugins/ethdoc/src/ethdoc.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import {
+  CompilationResult,
+  CompilatedContract,
```
Comment (yann300) [inline examples/plugins/ethdoc/src/ethdoc.ts]: there's a little typo here, this is `compiledContract`

File: examples/plugins/ethdoc/src/ethdoc.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  DeveloperDocumentation,
```
Comment (yann300) [inline examples/plugins/ethdoc/src/ethdoc.ts]: All these types are not really related to `remix-plugin` , we could easily have an another NPM package for referencing all the solidity related types ( I recall that something we already being discussed by the solidity team right?)

File: src/api/compiler/type/input.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  /** disabled by default */
+  enable: boolean
+  /**
+   * Optimize for how many times you intend to run the code.
+   * Lower values will optimize more for initial deployment cost, higher values will optimize more for high-frequency usage.
+   */
+  runs: number
+}
+
+export interface CompilaterMetadata {
```
Comment (yann300) [inline src/api/compiler/type/input.ts]: typo `CompiledMetadata`

File: examples/plugins/ethdoc/src/ethdoc.ts (hunk 2 above)
Comment (GrandSchtroumpf) [reply examples/plugins/ethdoc/src/ethdoc.ts]: We were thinking about putting them into the solc-js repository. But I'll need to change several things to their repo before. So it might take more time.

File: src/api/compiler/type/input.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    | 'spuriousDragon'
+    | 'byzantium'
+    | 'constantinople'
+  /** Metadata settings */
+  metadata?: CompilaterMetadata
+  /** Addresses of the libraries. If not all libraries are given here, it can result in unlinked objects whose output data is different. */
+  libraries: CompilerLibrarie
+}
+
+export interface CompilaterOptimizer {
```
Comment (yann300) [inline src/api/compiler/type/input.ts]: typo `CompiledOptimizer`

File: src/api/compiler/type/input.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  evmVersion:
+    | 'homestead'
+    | 'tangerineWhistle'
+    | 'spuriousDragon'
+    | 'byzantium'
+    | 'constantinople'
+  /** Metadata settings */
+  metadata?: CompilaterMetadata
+  /** Addresses of the libraries. If not all libraries are given here, it can result in unlinked objects whose output data is different. */
+  libraries: CompilerLibrarie
```
Comment (yann300) [inline src/api/compiler/type/input.ts]: typo `CompiledLibraries`

File: src/api/compiler/type/input.ts (hunk 3 above)
Comment (GrandSchtroumpf) [reply src/api/compiler/type/input.ts]: thx
